sdk/appcenter-auth/src/main/java/com/microsoft/appcenter/auth/Auth.java

```diff
@@ -461,24 +462,26 @@ private synchronized void initAuthenticationClient(String configurationPayload)
             JSONObject configuration = new JSONObject(configurationPayload);
             String identityScope = configuration.getString(IDENTITY_SCOPE);
             String authorityUrl = null;
+            String type = null;
             JSONArray authorities = configuration.getJSONArray(AUTHORITIES);
             for (int i = 0; i < authorities.length(); i++) {
                 JSONObject authority = authorities.getJSONObject(i);
                 if (authority.optBoolean(AUTHORITY_DEFAULT) && AUTHORITY_TYPE_B2C.equals(authority.getString(AUTHORITY_TYPE))) {
+                    type = AUTHORITY_TYPE_B2C;
                     authorityUrl = authority.getString(AUTHORITY_URL);
-                    break;
+                } else if (authority.optBoolean(AUTHORITY_DEFAULT) && AUTHORITY_TYPE_AAD.equals(authority.getString(AUTHORITY_TYPE))) {
```

Comment:
Could factor the check for default even if that increase indentation a little bit.